test/e2e/kubebuildercontext.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+)
+
+type CmdContext struct {
+	// environment variables in k=v format.
+	Env []string
+	Dir string
+}
+
+// KbContext specified to run e2e tests
+type KbContext struct {
+	*CmdContext
+	Domain    string
+	Group     string
+	Version   string
+	Kind      string
+	Resources string
+	ImageName string
+	Kubectl   *Kubectl
+}
+
+// ContextWithSuffix init with a random suffix for test KbContext stuff,
+// to avoid conflict when running tests synchronously.
+func ContextWithSuffix(testSuffix string, env []string) (*KbContext, error) {
+	testGroup := "bar" + testSuffix
+	path, err := filepath.Abs("e2e-" + testSuffix)
+	if err != nil {
+		return nil, err
+	}
+
+	cc := &CmdContext{
+		Env: env,
+		Dir: path,
+	}
+
+	return &KbContext{
+		Domain:     "example.com" + testSuffix,
+		Group:      testGroup,
+		Version:    "v1alpha1",
+		Kind:       "Foo" + testSuffix,
+		Resources:  "foo" + testSuffix + "s",
+		ImageName:  "e2e-test/controller-manager:" + testSuffix,
+		CmdContext: cc,
+		Kubectl: &Kubectl{
+			Namespace:  fmt.Sprintf("e2e-%s-system", testSuffix),
+			CmdContext: cc,
+		},
+	}, nil
+}
+
+func (cc *CmdContext) RunCommand(cmd *exec.Cmd) ([]byte, error) {
```

Comment:
Can we move this at the bottom so that testContext methods are together ?

Reply:
We can probably call this method just `Run` because it is already on `cmdContext` ?